PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/UseStorageAPIsWithLengthParameter.java

```diff
@@ -0,0 +1,44 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.*;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.List;
+
+
+public class UseStorageAPIsWithLengthParameter extends LocalInspectionTool {
+    @Nls
+    @NotNull
+    @Override
+    public String getDisplayName() {
+        return "Use Storage APIs With Length Parameter";
+    }
+
+    private static final List<String> METHODS_TO_CHECK = Arrays.asList("uploadFromFile", "upload", "uploadWithResponse");
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new JavaRecursiveElementWalkingVisitor() {
+            @Override
+            public void visitMethodCallExpression(PsiMethodCallExpression expression) {
+                super.visitMethodCallExpression(expression);
```

Comment:
Why are we calling the `super.visitMethodCallExpression` method? (usually these methods are just empty methods from the base class but not sure about this.)

Reply:
I understood that it continues traversing the AST before the method is exited. I saw it in all code samples so I figured its important.